code/modules/reagents/chemistry/machinery/chem_heater.dm

```diff
@@ -35,13 +35,14 @@
 
 /obj/machinery/chem_heater/Initialize()
 	. = ..()
-	create_reagents(200, NO_REACT)//Lets save some calculations here		
+	create_reagents(200, NO_REACT)//Lets save some calculations here
 	//TODO: comsig reaction_start and reaction_end to enable/disable the UI autoupdater - this doesn't work presently as there's a hard divide between instant and processed reactions
-	
+
 /obj/machinery/chem_heater/Destroy()
 	if(beaker)
 		UnregisterSignal(beaker.reagents, COMSIG_REAGENTS_REACTION_STEP)
-		QDEL_NULL(beaker)
+		beaker.forceMove(drop_location())
+		beaker = null
```

Comment:
This should be done in deconstruct(). Destroy() should still QDEL_NULL beaker if it's still there

Reply:
Is this okay?

Reply:
Yes, but bring back the `Destroy` that calls `QDEL_NULL(beaker)`, in the cases of deconstruct not being called.

Reply:
Destroy does call QDEL_NULL(beaker)? Am I misunderstanding?

Reply:
If you look at Files Changed it doesn't show that.

Reply:
That's how it handles it currently so I didn't change that.